src/Microsoft.DotNet.Cli.Utils/EmbedAppNameInHost.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Globalization;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    /// <summary>
+    /// Embeds the App Name into the AppHost.exe
+    /// </summary>
+    public static class EmbedAppNameInHost
```

Comment:
nit: while a decent name for an MSBuild task, not the best name for a general purpose class, in my opnion (assuming we're keeping this copy).

Perhaps `AppHostEmbedder` with a method called `EmbedTo`, perhaps taking `FilePath` (path to apphost), `FilePath` (destination file path, not directory), and `string` (path to embed), and returning void?

Reply:
I want to make the merge of the task and this code easier(after merge CLI and SDK repo). And this code also don't have test that is easy to port to CLI, so I want the change to be minimal.

Reply:
The code is in Utils and Utils don't have any project reference. Use FilePath will pull in InternalAbstraction which I don't know if it is a good idea. I will use string for now